std-bits/base/src/main/java/org/enso/base/cache/ReloadDetector.java

```diff
@@ -0,0 +1,39 @@
+package org.enso.base.cache;
+
+import org.enso.base.polyglot.EnsoMeta;
+import org.graalvm.polyglot.Value;
+
+/**
+ * Detects that the reload button has been pressed.
+ *
+ * <p>.hasReloadOccurred() returns true if the reload button was pressed since the last call to
+ * .hasReloadOccurred().
+ *
+ * <p>This uses a weak reference (created in eval'd Enso code) that is set to null on reload.
+ */
+public class ReloadDetector {
+  // Weak reference that is set to null on reload.
+  private Value trigger;
+
+  public ReloadDetector() {
+    resetTrigger();
+  }
+
+  public boolean hasReloadOccurred() {
+    var reloadHasOccurred = trigger.invokeMember("get").isNull();
```

Comment:
This code was correct, but it is not anymore. In the IDE the value returned from `get` should be an Enso `Error` now after the [most recent changes](https://github.com/enso-org/enso/pull/11577#discussion_r1867063485).

E.g. one should check for `isException()` and not (only) for `isNull()`.

Reply:
Done